zerver/tests/test_events.py

```diff
@@ -2071,7 +2071,12 @@ def test_create_bot(self) -> None:
 
         def get_bot_created_checker(bot_type: str) -> Validator[object]:
             if bot_type == "GENERIC_BOT":
-                check_services = check_list(sub_validator=None, length=0)
+                # Generic bots don't really understand the concept of
+                # "services", so we just enforce that we get an empty list.
+                def check_empty_list(var_name: str, val: object) -> None:
+                    if val != []:
+                        raise AssertionError(f'generic bots should get [] for {var_name}')
+                check_services = cast(Validator[object], check_empty_list)
```

Comment:
Repeating from #backend just so we don’t forget: this unchecked `cast` is wrong (a `Validator` must return the validated object), and we don’t actually need a custom validator here.

```suggestion
                check_services = equals([])
```

There’s a second instance of this in the same file.

Reply:
yeah, I think `equals([])` is the way to go here (as @andersk suggested on chat).  and yeah "Hold on, there's another place" came about when I tried on my branch to then simplify `check_list`.